SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -400,10 +400,17 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 TArray<Worker_ComponentId> EntityFactory::GetComponentPresenceList(const TArray<FWorkerComponentData>& ComponentDatas)
 {
 	TArray<Worker_ComponentId> ComponentPresenceList;
-	ComponentPresenceList.SetNum(ComponentDatas.Num() + 1);
+	ComponentPresenceList.Reserve(ComponentDatas.Num() + 1);
 	for (int i = 0; i < ComponentDatas.Num(); i++)
 	{
-		ComponentPresenceList[i] = ComponentDatas[i].component_id;
+		// Skip entity completeness tags, as we do not want them to be a part of ComponentPresence, and thus be delegated.
+		// This would create false completeness for workers which happened to be delegated the tag components.
+		if (ComponentDatas[i].component_id >= SpatialConstants::FIRST_EC_COMPONENT_ID
```

Comment:
Mike made a helper for this in the class info manager in [this PR](https://github.com/spatialos/UnrealGDK/pull/2524/files#diff-07317e4696ba4273343ce0815317410a5a35e1b37bb8381186025c82a8e32bfeR563). Could we move that into spatial constants and reuse it here instead?

Reply:
Also- if we're going for a proper solution, I don't think this is any better than moving the tags to after the presence, as this function is only used when creating the initial presence list. Ignoring them when reading the presence component instead would mean that the logic for making sure they don't get into the ACL would be localised, and doesn't require anyone creating presence components according to any specific rules.

Reply:
I guess I am worried about something else reading the component presence, but if it's just the one function, I guess it's okay to move the logic there.